app/lib/katello/util/path_with_substitutions.rb

```diff
@@ -14,7 +14,7 @@ def initialize(path, substitutions)
         @substitutions = substitutions
         @path = path
         @resolved = []
-        generate_substitutions_from_path if @substitutions.empty?
+        generate_substitutions_from_path if @substitutions.empty? && substitutions_needed.empty?
```

Comment:
How would this handle if you had a url like:   /content/$releasever/x86_64/,  wouldn't that fail to extract the substitutions?   Would it make more sense to check for the presence  of $basearch  specifically?

Reply:
The new logic means it tries to generate substitutions when it both a) does not have any substitution variables in the name and b) is not passed with any substitutions to fill variables, so `/content/$releasever/x86_64/` would either need a substitution passed with it or it would error like normally.

What would you want to check regarding `$releasever`? This should be handled as normal with RHEL 7 and under paths

Reply:
I think, if i'm reading this correct, if a url such as ` /content/$releasever/x86_64/` was used, substitutions_needed wouldn't be empty, so `generate_substitutions_from_path` wouldn't get called, right?